src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -319,15 +324,23 @@ private static boolean notFunctionMethodClass(HaxeComponentType type) {
     if (expression instanceof HaxeCallExpression) {
       final HaxeExpression callee = ((HaxeCallExpression)expression).getExpression();
       text = callee.getText();
+    } else if (expression instanceof HaxeRegularExpression) {
+      text = "REGEX";
+    } else if (expression instanceof HaxeLiteralExpression) {
+      text = "CONST_" + expression.getText().toUpperCase();
     }
 
     if (text != null) {
       candidates.addAll(HaxeNameSuggesterUtil.generateNames(text));
     }
+    if (candidates.isEmpty()) {
+      candidates.add("x");
+    }
 
     // todo: add suggestions
 
     final Set<String> usedNames = HaxeRefactoringUtil.collectUsedNames(expression);
+    usedNames.addAll(HaxeRefactoringUtil.collectKeywords());
```

Comment:
This avoids us using Haxe keywords as variable names.  (Which also happens to make parsing fail, and other bad things happen because we introduce an error block.)